client/src/app/s-degree-plan-editor/s-degree-plan-editor.component.ts

```diff
@@ -19,8 +21,31 @@ export class SDegreePlanEditorComponent implements OnInit {
    * (semester) isAddingSemester: Boolean, flag to check if user wants to add a semester
    */
 
-  constructor() {
-    /** Dummy data for plan used to demo markup */
+  constructor(
+    private planService: PlanService,
+    private userService: UserService
+  ) {
+    this.planService.formatPlan().subscribe((result) => {
+      this.plan = result;
+      this.plan.user = "";
+      this.plan.program = "";
+      this.plan.isAddingYear = false;
+      this.plan.years.forEach(
+        (semester) => (semester.isAddingSemester = false)
+      );
```

Comment:
Making the user data look like the dummy data so we can readily use it in the markup